Implemented itemCounts function

JoodyKonno opened the pull request:
The itemCounts takes an array of letters and returns a Map of letters with the number of occurrences of that letter o the letters array.

Example: 

`
itemCounts(['a', 'a', 'a', 'b', 'c', 'c']);
// it returns { 'a': 3, 'b': 1,' c': 2 }
`

Commit:
Added require.main === module condition for manual testing

In order to prevent the manual testing running when imported as application modules, a condition was added. Now it only runs when execu
ted directly.

For execute with manual testing run on terminal:

- node textalyze.js
or
- npm start

Commit:
Remove unnecessary parseInt from itemCounts

The parseInt() was being used on a variable that was created inside the same function with an integer value. Removing it may not cause any side effects

Commit:
Refactor the array foreach to an array Reduce on itemCounts

The itemCounts could be written on a single reduce function which reduces the complexity and number of variables inside the function

jfarmer commented on textalyze.js:
```diff
@@ -11,24 +11,14 @@
-}
+const itemCounts = array => array
+  .reduce((counts, item) => {
+    !counts.has(item)
```
Nice!

Check out what [Map.prototype.set](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map/set) returns, though.  It returns the updated `Map` object directly.

So:

```javascript
!counts.has(item) ? counts.set(item, 1) : counts.set(item, counts.get(item) + 1);
return counts
```

Can become

```javascript
return !counts.has(item) ? counts.set(item, 1) : counts.set(item, counts.get(item) + 1);
```

jfarmer replied:
BTW, unless I say something needs to be fixed, feel free to merge the PR into your code.  I will always find a way to give feedback.  You can submit a new PR for the refactoring and request feedback on that, too.

Commit:
Reduced the reduce function to a single statement

The Map.set() function already returns the Map object, so the return statement was redundant.
